recovery/cmd/backup_size.go

```diff
@@ -0,0 +1,188 @@
+	usrLocalPath        = "/usr/local"
+	kubeletPath         = "/var/lib/kubelet"
+	etcPath             = "/etc"
+	binPath             = "/etc/kubernetes/static-pod-resources/bin/"
+	backupDir           = "/var/recovery"
+	backupSizeSafetyNet = 10.00 // in GiB
+	units               = []string{"B", "KiB", "MiB", "GiB", "TiB", "EiB", "ZiB"}
+)
+
+type resource struct {
+	dirPath string
+}
+
+type resourceList struct {
+	resources *[]resource
+}
+
+// compareBackupToDisk verifies disk space required against available
+// disk space and returns boolean value
+func compareBackupToDisk() (bool, error) {
+
+	estimated, err := estimateDiskSize()
+	if err != nil {
+		log.Errorf("Couldn't calculate estimated disk space")
+		return false, err
+	}
+
+	freeDisk, err := diskPartitionSize()
+	if err != nil {
+		log.Errorf("Couldn't calculate free disk space")
+		return false, err
+	}
+
+	estimated, ebytes := sizeConversion(estimated)
+	freeDisk, fbytes := sizeConversion(freeDisk)
+
+	log.Infof("Available disk space : %.2f %s; Estimated disk space required for backup: %.2f %s \n", freeDisk, fbytes, estimated, ebytes)
+
+	// find the index of estimated and freedisk space in units[]
+	var ebytesID, fbytesID int
+	for i := range units {
+		if units[i] == ebytes {
+			ebytesID = i
+		}
+		if units[i] == fbytes {
+			fbytesID = i
+		}
+	}
+
+	if ebytesID <= fbytesID {
+		x := float64(fbytesID - ebytesID)
+		if x != 0.0 {
+			estimated /= 1024 * x
+		}
+		if freeDisk > backupSizeSafetyNet+estimated {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+// estimateDiskSize calculate the required backup size
+// returns: disk size(int), error
+func estimateDiskSize() (float64, error) {
+
+	DirList := resourceList{
+		&[]resource{{staticPodsPath},
+			{clusterPath},
+			{usrLocalPath},
+			{kubeletPath},
+			{etcPath},
+		},
+	}
+
+	estDirMap := map[string]float64{}
+	var total float64
+
+	for _, v := range *DirList.resources {
+		_, err := os.Lstat(v.dirPath)
+		if err != nil {
+			return total, err
+		}
+
+		switch v.dirPath {
+		case staticPodsPath:
+			binPathSize := dirSize(binPath)
+			estDirMap[v.dirPath] = dirSize(v.dirPath) - binPathSize
+
+		case clusterPath:
+			estDirMap[v.dirPath] = dirSize(v.dirPath) + estDirMap[staticPodsPath]
+
+		default:
+			estDirMap[v.dirPath] = dirSize(v.dirPath)
+		}
+
+		total += estDirMap[v.dirPath]
+
+	}
+	return total, nil
+}
+
+// diskPartitionSize calculate current disk space
+// returns:     disk size(int), error
+func diskPartitionSize() (float64, error) {
+
+	var (
+		stat          unix.Statfs_t
+		freeDiskSpace float64
+	)
+
+	err := unix.Statfs(backupDir, &stat)
+	if err != nil {
+		return freeDiskSpace, err
+	}
+
+	freeDiskSpace = float64(stat.Bavail * uint64(stat.Bsize))
+
+	err = unix.Statfs("/", &stat)
+	if err != nil {
+		return freeDiskSpace, err
+	}
+	total, tbytes := sizeConversion(float64(stat.Blocks * uint64(stat.Bsize)))
+	log.Infof("Total disk size is: %.2f %s", total, tbytes)
```

Comment:
I assume this is here for informational purposes only? The size of "/" wouldn't be the "Total disk size", though - it's just the total size of the rootfs, which would be irrelevant to /var/recovery. Either way, I'm not sure why we'd need to log the total disk size

Reply:
yes, "/" is considered as rootfs, in any case, it is excluding "/boot" mountpoint since we can't use that. This is only for information purpose, it is not needed to calculate or find the free and/or estimated space.

Reply:
My point, though, is that this would not be relevant to /var/recovery (particularly if either /var/recovery or /var is a separate partition), and is not "Total disk size". It is just the size of the root filesystem itself, not the disk

Reply:
Removed with 74f7f6eca77c0b69ab1530c7ac44eb2c204a8f8b